CONTRIBUTING.md

```diff
@@ -1,24 +1,430 @@
+
+To learn more about running tests with Swift Package Manager, visit the
+[Testing](#testing) section.
+
+<!-- SwiftPM troubleshooting -->
+<!-- TODO: Common issues and fixes like resolve depencies & reset cache. -->
+
+#### **[CocoaPods]**
+[CocoaPods] is another popular dependency manager used in Apple development.
+Firebase supports development with CocoaPods 1.10.0 (or later). If you choose to
+develop using CocoaPods, it's recommend to use
+[`cocoapods-generate`][cocoapods-generate], a plugin that generates a
+[workspace] from a [podspec]. This plugin allows you to quickly generate a
+development workspace using any library's podspec. All of the podspecs for
+Firebase's libraries are located in the repo's root directory.
+
+#### Installation
+- **[CocoaPods]**
+  <!-- This line is intentionally left blank. -->
+  To check if your machine has CocoaPods installed, run `pod --version` in
+  terminal. If the command fails with a `command not found` error, then you'll
+  need to install CocoaPods.
+
+  To install, please refer to CocoaPods's [Getting Started][cocoapods-install] guide.
+
+- **[cocoapods-generate]**
+  <!-- This line is intentionally left blank. -->
+  Please see [cocoapods-generate] for instructions on how to install.
+
+#### Developing with CocoaPods
+With **CocoaPods** and **cocoapods-generate** installed, the `pod gen` command
+makes it easy to develop specific Firebase libraries.
+
+```console
+pod gen Firebase{name here}.podspec --local-sources=./ --auto-open --platforms=ios
+```
+
+- If the CocoaPods cache is out of date, you may need to run
+  `pod repo update` before the `pod gen` command.
+-  Set the `--platforms` option to `macos` or `tvos` to develop on those
+   platforms. Since 10.2, Xcode does not properly handle multi-platform
+   CocoaPods workspaces.
+
+<details>
+<summary><i>Developing for Mac Catalyst?</i></summary>
+
+<!-- The above line is intentionally left blank. -->
+To develop for [Mac Catalyst], there are a few additional steps to configure
+the project.
+
+1. Run `pod gen {name here}.podspec --local-sources=./ --auto-open --platforms=ios`
+2. Check the **Mac** box in the host app's **Build Settings**
+3. Sign the host app in the **Signing & Capabilities** tab
+4. Navigate to **Pods** in the **Project Manager**
+5. Add **Signing** to the **host app** and **unit test** targets
+6. Select the **Unit-unit** scheme
+7. **Run** it to build and test
+
+**Alternatively**, disable signing in each target:
+1. Go to **Build Settings** tab
+2. Click **+**
+3. Select **Add User-Defined Setting**
+4. Add `CODE_SIGNING_REQUIRED` setting with a value of `NO`
+</details>
+<br> <!-- This new line is for styling purposes. -->
+
+<!-- #### **Sample Apps** -->
+
+#### **Style Guide**
+This code in this repo is styled in accordance to `clang-format` conventions.
+
+#### Styling your code
+The [./scripts/style.sh] script makes it easy to style your
+code during development.
```

Comment:
I'd add that running style.sh on just the folder you worked in is th emost efficient (in terms of runtime, *and* in making sure you only format your own changes).